scheduling/allocator.rb

```diff
@@ -597,7 +610,15 @@ def map_volumes_to_devices
 
     def create_storage_volumes(vm, vm_host)
       @request.storage_volumes.each do |disk_index, volume|
-        spdk_installation_id = StorageAllocation.allocate_spdk_installation(vm_host.spdk_installations)
+        if vm_host.vhost_block_backends_dataset.exclude(allocation_weight: 0).empty?
+          spdk_installation_id = StorageAllocation.allocate_spdk_installation(vm_host.spdk_installations)
+          vhost_block_backend_id = nil
```

Comment:
If you want to include this line to be explicit, that's fine.  Just note that it isn't required.  It will automatically be nil if this branch is taken, because the other branch attempts to define it:

```
$ ruby -e 'if ARGV.empty?; a = 1; else b = 1; end; p [a,b]'
[1, nil]

ruby -e 'if ARGV.empty?; a = 1; else b = 1; end; p [a,b]' foo
[nil, 1]
```

Same with `spdk_installation_id` in the else block.